confluence-slack-integration/confluence-slack-server-integration-plugin/src/main/java/com/atlassian/confluence/plugins/slack/spacetochannel/api/notifications/DefaultAttachmentBuilder.java

```diff
@@ -79,21 +79,25 @@ public Attachment buildAttachment(final SpaceContentEntityObject content) {
     }
 
     private String calculateFooterTimestampText(final SpaceContentEntityObject content) {
+        final ConfluenceUser creator = content.getCreator();
         final String slackPublishedTimestamp = i18nResolver.getText(
```

Comment:
how about having a method
getSlackTimeStamp(String resolver, ConfluenceUser user, Date date)

Just to avoid the repetition of code we can use at both the places **slackUpdatedTimestamp** and **slackPublishedTimestamp**

Reply:
Such method would invoke just one method - `i18nResolver.getText()` - with different parameters. Such low-level optimization just doesn't seem reasonable. Some code duplication would still be avoided, but instead readability would worsen slightly because of one more abstraction layer added.